pkg/build/strategies/layered/layered.go

```diff
@@ -35,11 +36,15 @@ func New(config *api.Config, scripts build.ScriptsHandler, overrides build.Overr
 	if err != nil {
 		return nil, err
 	}
+	tarHandler := tar.New()
+	if config.ExcludeRegExp != tar.DefaultExclusionPattern.String() {
```

Comment:
I was thinking the check was cheaper than the construction of the regexp, but sure, both are cheap most likely... old habits :-) ... will remove / simplify code